pkg/libinit/proc_linux.go

```
@@ -41,6 +48,33 @@ func WithTTYControl(ctty bool) CommandModifier {
 	}
 }
 
+func WithMultiTTY(mtty bool, openFn func([]string) ([]*os.File, error), ttyNames []string) CommandModifier {
+	return func(c *exec.Cmd) {
+		if mtty {
+			ww, err := openFn(ttyNames)
+			if err != nil {
```

you'll need to remove the print, it's a package. You can return an error however.

---

It return a function - if we would want to return an error we would need to modify the `mod` function as well to be able to return an error. I did not wanted to change the flow here - so I'd just return.

---

so how about this. You want to use all the available consoles, which will fail if you return early if just one fails. 

You might use errors.Join accumulate a list of errors, and accumulate your list of consoles too. 

Is the problem that the caller of the modifier will quit with a non-null error?

It would be nice if you could return with 
zero or more consoles, error
the caller can print the error, if it is non-nil, and use whatever consoles are returned.

This is similar to what io readers do: they can return data AND an error and let you pick what to do.

This is one of those cases where you want to know if something happened, but you want to try all the things. 

I hate to see packages just print stuff to stderr because it makes it so hard to read the GitHub CI output when all these random messages appear.

---

I agree that it makes generally sense to catch potential errors here but that would require a larger rework as we return a `CommandModifier` which currently is defined as `type CommandModifier func(c *exec.Cmd)`. We would need to modify that type to also return errors which means changing every command modifier that currently exist as well. That's why Chris didn't want to change a lot here as the scope of the PR would be much broader then.